### PR TITLE
Switch logic for injection

### PR DESCRIPTION
**Pull request by howardjohn**
Implement new injector logic, doc: https://docs.google.com/document/d/1umIDCxvTwfOBtNh-bbRb3eyOxJ_4jsSBXl4pfcAYqR0/edit?resourcekey=0-OOCyBAXemTcnpyDRjU48pQ#heading=h.xw1gqgyqs5b

### CODE REVIEW

File: manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml [hunk 1]
```diff
@@ -181,7 +181,8 @@ data:
           "enabled": true
         },
         "rewriteAppHTTPProbe": true,
-        "templates": {}
+        "templates": {},
+        "useLegacySelectors": false
```
Comment (costinm) [inline manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml:185]: Do we want it false by default ?

File: manifests/charts/istio-control/istio-discovery/templates/mutatingwebhook.yaml [hunk 2]
```diff
@@ -15,71 +39,151 @@ metadata:
...
-        operator: DoesNotExist
-{{- else if .Values.revision }}
-      matchExpressions:
-      - key: istio-injection
-        operator: DoesNotExist
-      - key: istio.io/rev
-        operator: In
-        values:
-        - {{ .Values.revision }}
+{{- if .Values.sidecarInjectorWebhook.useLegacySelectors}}
```
Comment (costinm) [inline manifests/charts/istio-control/istio-discovery/templates/mutatingwebhook.yaml:42]: Can we have the useLegacySelectors in mesh config ?
I don't remember where sidecarInjectWebhook is coming from...

File: manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml (hunk 1 above)
Comment (howardjohn) [reply manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml:185]: true by default in master (1.10), false in 1.9 was what we discussed. I will change to false in the backport

File: manifests/charts/istio-control/istio-discovery/templates/mutatingwebhook.yaml (hunk 2 above)
Comment (howardjohn) [reply manifests/charts/istio-control/istio-discovery/templates/mutatingwebhook.yaml:42]: What is the benefit? I thought we decided that Kubernetes things would be configured by helm api, and runtime components by mesh config. This is directly configuring a k8s object (MutatingAdmissionWebhook).

sidecarInjectWebhook is in values.yaml

Additionally, this is a feature flag - not a proper API - and will be removed in 1.11

**Review by costinm — APPROVED** [review]
I have some doubts about where to put the new config - but since it's temporary we can go ahead.

File: releasenotes/notes/injector-selectors.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    * There is no way to opt-in an individual pod, with `sidecar.istio.io/inject`, without enabling injection
+      for the entire namespace.
+
+    These limitations have both been resolved. As a result, additional pods may be injected that were not in previous versions,
+    if they exist in a namespace without an `istio-injection` label set but have the `sidecar.istio.io/inject` label set on the pod.
+    This is expected to be an uncommon case, so for most users there will be no behavioral changes to existing pods.
+
+    If this behavior is not desired, it can be temporarily disabled with `--set values.sidecarInjectorWebhook.useLegacySelectors=true`;
+    this option will be removed in future releases.
```
Comment (ericvn) [inline releasenotes/notes/injector-selectors.yaml:28]: Maybe break this sentence at the ;. Replace ; with a period and capitalize This.

Comment (howardjohn) [reply releasenotes/notes/injector-selectors.yaml]: done

Comment (ericvn) [reply releasenotes/notes/injector-selectors.yaml:28]: John, I don't see that the change was actually made.

**Review by ericvn — APPROVED** [review]

**Pull request merged**